### ISSUE TITLE
MarGo build failed

### ISSUE THREAD

**Issue by diannewood** [DisposaBoy/GoSublime#760]
I am a licensed user and recently moved to a new machine. I noticed that I have several gosublime.margo processes eating up my cpu. I uninstalled sublime but they are still running. When I reinstalled I got this out:

| GoSublime error: MarGo build failed
|                > This is possibly a bug or miss-configuration of your environment.
|                > For more help, please file an issue with the following build output
|                > at: https://github.com/DisposaBoy/GoSublime/issues/new
|                > or alternatively, you may send an email to: gosublime@dby.me
|                > 
|                > 
|                > cmd: `['/usr/local/go/bin/go', 'build', '-tags', '', '-v', '-o', 'gosublime.margo_r17.02.16-1_go?.exe', 'disposa.blue/cmd/margo']`
|                > stdout: `
|                > 
|                > `
|                > stderr: `
|                > disposa.blue/margo
|                > disposa.blue/something-borrowed/types
|                > net
|                > disposa.blue/margo/meth/importpaths
|                > # net
|                > fatal error: unexpected signal during runtime execution
|                > [signal 0xb code=0x1 addr=0xb01dfacedebac1e pc=0xf0eb]
|                > 
|                > runtime stack:
|                > runtime.throw(0x4971c0, 0x2a)
|                > 	/usr/local/go/src/runtime/panic.go:547 +0x90
|                > runtime.sigpanic()
|                > 	/usr/local/go/src/runtime/sigpanic_unix.go:12 +0x5a
|                > runtime.unlock(0x982540)
|                > 	/usr/local/go/src/runtime/lock_sema.go:107 +0x14b
|                > runtime.(*mheap).alloc_m(0x982540, 0x1, 0x12, 0xe2d6c8)
|                > 	/usr/local/go/src/runtime/mheap.go:492 +0x314
|                > runtime.(*mheap).alloc.func1()
|                > 	/usr/local/go/src/runtime/mheap.go:502 +0x41
|                > runtime.systemstack(0xc8204a7e58)
|                > 	/usr/local/go/src/runtime/asm_amd64.s:307 +0xab
|                > runtime.(*mheap).alloc(0x982540, 0x1, 0x10000000012, 0xed8f)
|                > 	/usr/local/go/src/runtime/mheap.go:503 +0x63
|                > runtime.(*mcentral).grow(0x983ff0, 0x0)
|                > 	/usr/local/go/src/runtime/mcentral.go:209 +0x93
|                > runtime.(*mcentral).cacheSpan(0x983ff0, 0xdf75f0)
|                > 	/usr/local/go/src/runtime/mcentral.go:89 +0x47d
|                > runtime.(*mcache).refill(0xaf4000, 0x12, 0xdf75f0)
|                > 	/usr/local/go/src/runtime/mcache.go:119 +0xcc
|                > runtime.mallocgc.func2()
|                > 	/usr/local/go/src/runtime/malloc.go:642 +0x2b
|                > runtime.systemstack(0xc820024a00)
|                > 	/usr/local/go/src/runtime/asm_amd64.s:291 +0x79
|                > runtime.mstart()
|                > 	/usr/local/go/src/runtime/proc.go:1051
|                > 
|                > goroutine 1 [running]:
|                > runtime.systemstack_switch()
|                > 	/usr/local/go/src/runtime/asm_amd64.s:245 fp=0xc822074818 sp=0xc822074810
|                > runtime.mallocgc(0x120, 0x42e1a0, 0x0, 0xc822afde60)
|                > 	/usr/local/go/src/runtime/malloc.go:643 +0x869 fp=0xc8220748f0 sp=0xc822074818
|                > runtime.newobject(0x42e1a0, 0x0)
|                > 	/usr/local/go/src/runtime/malloc.go:781 +0x42 fp=0xc822074918 sp=0xc8220748f0
|                > cmd/compile/internal/gc.Prog(0x31ba, 0xc822afeb40)
|                > 	/usr/local/go/src/cmd/compile/internal/gc/gsubr.go:117 +0x58 fp=0xc822074980 sp=0xc822074918
|                > cmd/compile/internal/amd64.gins(0x31ba, 0xc822afea20, 0xc822afeb40, 0x4c213e)
|                > 	/usr/local/go/src/cmd/compile/internal/amd64/gsubr.go:636 +0xf6 fp=0xc822074a60 sp=0xc822074980
|                > cmd/compile/internal/amd64.gmove(0xc822afea20, 0xc822afeb40)
|                > 	/usr/local/go/src/cmd/compile/internal/amd64/gsubr.go:540 +0x501 fp=0xc822074be0 sp=0xc822074a60
|                > cmd/compile/internal/amd64.gmove(0xc822afea20, 0xc822afe990)
|                > 	/usr/local/go/src/cmd/compile/internal/amd64/gsubr.go:560 +0x350 fp=0xc822074d60 sp=0xc822074be0
|                > cmd/compile/internal/gc.componentgen_wb.func4(0xc8200905a0, 0x8, 0x1ae701)
|                > 	/usr/local/go/src/cmd/compile/internal/gc/gen.go:1147 +0x285 fp=0xc822074e08 sp=0xc822074d60
|                > cmd/compile/internal/gc.visitComponents(0xc820090f00, 0x0, 0xc822074f70, 0xc822074f01)
|                > 	/usr/local/go/src/cmd/compile/internal/gc/gen.go:1199 +0x5a1 fp=0xc822074e48 sp=0xc822074e08
|                > cmd/compile/internal/gc.componentgen_wb(0xc822aede60, 0xc8215dcfc0, 0xc822af9300, 0x0)
|                > 	/usr/local/go/src/cmd/compile/internal/gc/gen.go:1149 +0xa1c fp=0xc822074fc0 sp=0xc822074e48
|                > cmd/compile/internal/gc.sgen_wb(0xc822aede60, 0xc8215dcfc0, 0x10, 0x0)
|                > 	/usr/local/go/src/cmd/compile/internal/gc/cgen.go:2274 +0x459 fp=0xc8220750b8 sp=0xc822074fc0
|                > cmd/compile/internal/gc.cgen_wb(0xc822aede60, 0xc8215dcfc0, 0x0)
|                > 	/usr/local/go/src/cmd/compile/internal/gc/cgen.go:85 +0x427 fp=0xc822075488 sp=0xc8220750b8
|                > cmd/compile/internal/gc.Cgen_as_wb(0xc8215dcfc0, 0xc822aede60, 0x0)
|                > 	/usr/local/go/src/cmd/compile/internal/gc/gen.go:935 +0x166 fp=0xc8220754e8 sp=0xc822075488
|                > cmd/compile/internal/gc.Cgen_as(0xc8215dcfc0, 0xc822aede60)
|                > 	/usr/local/go/src/cmd/compile/internal/gc/gen.go:886 +0x30 fp=0xc822075508 sp=0xc8220754e8
|                > cmd/compile/internal/gc.gen(0xc822af6990)
|                > 	/usr/local/go/src/cmd/compile/internal/gc/gen.go:833 +0x191 fp=0xc822075660 sp=0xc822075508
|                > cmd/compile/internal/gc.Genlist(0xc8201bd4e0)
|                > 	/usr/local/go/src/cmd/compile/internal/gc/gen.go:215 +0x30 fp=0xc822075670 sp=0xc822075660
|                > cmd/compile/internal/gc.gen(0xc822af6a20)
|                > 	/usr/local/go/src/cmd/compile/internal/gc/gen.go:655 +0x41a fp=0xc8220757c8 sp=0xc822075670
|                > cmd/compile/internal/gc.Genlist(0xc8201bcfc0)
|                > 	/usr/local/go/src/cmd/compile/internal/gc/gen.go:215 +0x30 fp=0xc8220757d8 sp=0xc8220757c8
|                > cmd/compile/internal/gc.compile(0xc820cfa7e0)
|                > 	/usr/local/go/src/cmd/compile/internal/gc/pgen.go:484 +0xc02 fp=0xc822075a48 sp=0xc8220757d8
|                > cmd/compile/internal/gc.funccompile(0xc820cfa7e0)
|                > 	/usr/local/go/src/cmd/compile/internal/gc/dcl.go:1450 +0x1c0 fp=0xc822075ac0 sp=0xc822075a48
|                > cmd/compile/internal/gc.Main()
|                > 	/usr/local/go/src/cmd/compile/internal/gc/lex.go:472 +0x2116 fp=0xc822075de0 sp=0xc822075ac0
|                > cmd/compile/internal/amd64.Main()
|                > 	/usr/local/go/src/cmd/compile/internal/amd64/galign.go:127 +0x58d fp=0xc822075e48 sp=0xc822075de0
|                > main.main()
|                > 	/usr/local/go/src/cmd/compile/main.go:32 +0x395 fp=0xc822075f20 sp=0xc822075e48
|                > runtime.main()
|                > 	/usr/local/go/src/runtime/proc.go:188 +0x2b0 fp=0xc822075f70 sp=0xc822075f20
|                > runtime.goexit()
|                > 	/usr/local/go/src/runtime/asm_amd64.s:1998 +0x1 fp=0xc822075f78 sp=0xc822075f70
|                > `
|                > exception: `None`
|   install state: busy
| sublime.version: 3126
| sublime.channel: stable
|       about.ann: a17.02.16-1
|   about.version: r17.02.16-1
|         version: r17.02.16-1
|        platform: osx-x64
|            ~bin: ~/Library/Application Support/Sublime Text 3/Packages/User/GoSublime/osx-x64/bin
|       margo.exe: ~bin/gosublime.margo_r17.02.16-1_go?.exe (missing)
|          go.exe: /usr/local/go/bin/go (ok)
|      go.version: go?
|          GOROOT: /usr/local/go
|          GOPATH: ~/go
|           GOBIN: (not set) (should usually be `(not set)`)
|       set.shell: []
|       env.shell: /bin/bash
|       shell.cmd: ['/bin/bash', '-l', '-c', '${CMD}']

**Comment by coftysweet** [comment]
I had the same problem but could resolve it. I upgraded go from 1.6 to 1.8 by overwriting the existing installation with the new one. It transpired that a (possible renamed?) file from the old installation remained (signal1_unix.go) which provided the same contents as new signal_unix.go. I renamed the full /usr/local/go folder and extracted the package again to /usr/local/go preventing other maybe renamed files. Now it works.
I hope this works for you too.
Best regards
coftysweet
